src/lib.rs

```diff
@@ -75,79 +72,93 @@ pub struct UPClientZenoh {
     query_map: QueryMap,
     // Save the callback for RPC response
     rpc_callback_map: RpcCallbackMap,
-    // My authority
-    authority_name: String,
+    // URI
+    uri: UUri,
 }
 
-impl UPClientZenoh {
-    /// Create `UPClientZenoh` by applying the Zenoh configuration, `UAuthority`.
+impl UPTransportZenoh {
+    /// Create `UPTransportZenoh` by applying the Zenoh configuration, local `UUri`.
     ///
     /// # Arguments
     ///
-    /// * `config` - Zenoh configuration. You can refer to [here](https://github.com/eclipse-zenoh/zenoh/blob/0.11.0-rc.3/DEFAULT_CONFIG.json5) for more configuration details.
-    /// * `authority_name` - The authority name. We need it to generate Zenoh key since authority might be omitted in `UUri`.
+    /// * `config` - Zenoh configuration. You can refer to [here](https://github.com/eclipse-zenoh/zenoh/blob/0.11.0/DEFAULT_CONFIG.json5) for more configuration details.
+    /// * `uri` - Local `UUri`. Note that the Authority of the `UUri` MUST be non-empty.
```

Comment:
FMPOV it would add to transparency if you would document here that the resource ID must be 0 and then actually return an error if it isn't. Otherwise, a non-zero resource ID would be silently ignored and the user might wonder why ...